api.go

```diff
@@ -33,37 +33,20 @@ func (a *API) HandleDirectory(dir string) error {
 		return fmt.Errorf("Cannot list clang-c directory: %v", err)
 	}
 
-	h := &HeaderFile{
-		api: a,
-
-		dir: dir,
-
-		lookupEnum:        map[string]*Enum{},
-		lookupNonTypedefs: map[string]string{},
-		lookupStruct: map[string]*Struct{
-			"cxstring": &Struct{
-				Name:  "cxstring",
-				CName: "CXString",
-			},
-		},
-	}
-
 	for _, hf := range headers {
-		name := hf.Name()
-
-		if hf.IsDir() || !strings.HasSuffix(name, ".h") {
+		if hf.IsDir() || !strings.HasSuffix(hf.Name(), ".h") {
 			continue
 		}
-
-		h.name = dir + name
+        
+        h := newHeaderFile(a, dir + hf.Name(), dir)
```

Comment:
I am wondering why HeaderFile.Name is here instiated with "dir + hf.Name()"? Why do we need the directory then at all?